channeldb/payment_control.go

```diff
@@ -187,9 +192,12 @@ func (p *PaymentControl) RegisterAttempt(paymentHash lntypes.Hash,
 // duplicate payments to the same payment hash. The provided preimage is
 // atomically saved to the DB for record keeping.
 func (p *PaymentControl) Success(paymentHash lntypes.Hash,
-	preimage lntypes.Preimage) error {
+	preimage lntypes.Preimage) (*route.Route, error) {
```

Comment:
To avoid having to do this change, one could instead provinde the route as a param to the `Success` method in the routing `ControlTower`.

Reply:
I am not sure if that is better. It puts a new requirement on the caller of `ControlTower`.